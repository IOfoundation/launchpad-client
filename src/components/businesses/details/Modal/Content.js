import React, {Fragment} from 'react';
import {PropTypes} from 'prop-types';

import DateSection from './DateSection';
import Section from './Section';
import PostedBy from './PostedBy';
import Link from './Link';

const Layout = props => {
  const {
    title,
    postedBy,
    address,
    link,
    description,
    closed,
    start,
    end,
  } = props;

  return (
    <Fragment>
      <i className="material-icons modal-events__close" onClick={closed}>
        {'close'}
      </i>
      <h1 className="modal-events__title">{title}</h1>
      <PostedBy url={postedBy.website} name={postedBy.name} />
      <DateSection start={start} end={end} />
      <Section content={address} modifiers={['extra-space']} />
      <Link link={link} />
      <Section content={description} />
    </Fragment>
  );
};

Layout.propTypes = {
  address: PropTypes.string,
  closed: PropTypes.func,
  description: PropTypes.string,
  end: PropTypes.string,
  link: PropTypes.string,
  postedBy: PropTypes.shape({
    name: PropTypes.string,
    website: PropTypes.string,
  }),
  start: PropTypes.string,
  title: PropTypes.string,
};

export default Layout;
