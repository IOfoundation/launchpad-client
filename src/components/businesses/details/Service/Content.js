import React from 'react';
import {PropTypes} from 'prop-types';

const Content = props => {
  const {title, content, children} = props;
  let $content = null;

  if (content || children) {
    $content = (
      <div className="service-content">
        <h3 className="service-content__title text-semi">{title}</h3>
        <div className="service-content__content">{content || children}</div>
      </div>
    );
  }

  return $content;
};

Content.propTypes = {
  content: PropTypes.string,
  title: PropTypes.string,
};

export default Content;
