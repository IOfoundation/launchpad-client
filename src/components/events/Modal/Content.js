import React, {Fragment} from 'react';
import {PropTypes} from 'prop-types';

import DateSection from '../../businesses/details/Modal/DateSection';
import Section from '../../businesses/details/Modal/Section';
import PostedBy from '../../businesses/details/Modal/PostedBy';
import Link from '../../businesses/details/Modal/Link';

const Content = props => {
  const {
    title,
    postedBy,
    address,
    link,
    description,
    closed,
    start,
    end,
  } = props;

  return (
    <Fragment>
      <i className="material-icons modal-events__close" onClick={closed}>
        {'close'}
      </i>
      <h1 className="modal-events__title">{title}</h1>
      <PostedBy url={postedBy.website} name={postedBy.name} />
      <DateSection start={start} end={end} />
      <Section content={address} modifiers={['extra-space']} />
      <Link link={link} />
      <Section content={description} />
    </Fragment>
  );
};

Content.propTypes = {
  address: PropTypes.string,
  closed: PropTypes.func,
  description: PropTypes.string,
  end: PropTypes.string,
  link: PropTypes.string,
  postedBy: PropTypes.shape({
    name: PropTypes.string,
    website: PropTypes.string,
  }),
  start: PropTypes.string,
  title: PropTypes.string,
};

export default Content;
