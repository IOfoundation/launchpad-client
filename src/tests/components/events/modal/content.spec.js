import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Content from '../../../../components/events/Modal/Content';

const closed = jest.fn();
const props = {
  address: 'Address',
  closed,
  description: 'Descripton',
  end: 'end',
  link: 'url',
  postedBy: {
    name: 'name',
    website: 'website',
  },
  start: 'Starts',
  title: 'Title',
};

describe('<Content />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Content {...props} />);
  });

  it('renders snapshot of Modal Content', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should trigger the close option', () => {
    const button = wrapper.find('.modal-events__close');

    button.simulate('click');

    expect(closed).toBeCalled();
  });
});
