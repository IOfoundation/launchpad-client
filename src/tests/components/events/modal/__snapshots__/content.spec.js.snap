// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> renders snapshot of Modal Content 1`] = `
<Fragment>
  <i
    className="material-icons modal-events__close"
    onClick={[MockFunction]}
  >
    close
  </i>
  <h1
    className="modal-events__title"
  >
    Title
  </h1>
  <PostedBy
    name="name"
    url="website"
  />
  <DateSection
    end="end"
    start="Starts"
  />
  <Section
    content="Address"
    modifiers={
      Array [
        "extra-space",
      ]
    }
  />
  <Link
    link="url"
  />
  <Section
    content="Descripton"
  />
</Fragment>
`;
